### PR TITLE
dev_v16 build and versions updates to support Hera/Orion

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
**Description**

This PR contains the following updates to support building the GFSv16.2.0 package via the `dev_v16` branch on Hera/Orion:

1. update Hera/Orion global-workflow modulefiles to use special `hpc-stack-gfsv16` hpc-stack installs created by hpc-stack team
2. change `python` to `anaconda` on Hera
3. remove module loads in Hera/Orion module_base modulefiles that won't be needed or won't be available on either machine: `libjpeg`, `bufr_dump`, `util_shared`, `perl` (will need to come back and check on `perl` on Hera, likely needed for GSI monitoring package, removing for now to avoid errors at loadtime)
4. build scripts updates for supporting R&D machines: `build_fv3nc2nemsio.sh`, `build_regrid_nemsio.sh`, `build_tropcy_NEMS.sh`
5. makefile updates for supporting R&D machines: `enkf_chgres_recenter_nc.fd/makefile`, `supvit.fd/makefile`, `syndat_getjtbul.fd/makefile`, `syndat_maksynrc.fd/makefile`, `syndat_qctropcy.fd/makefile`, `tave.fd/makefile`, `vint.fd/makefile`
6. checkout.sh updates for component versions: GSI (`gfsda.v16.2.0.1` @MichaelLueken-NOAA) and UPP (`upp_v8.1.1` @WenMeng-NOAA)
7. checkout.sh update to add WAFS checkout to the `-o` operations checkout mode; won't need or support WAFS outside of WCOSS2
8. update link_fv3gfs.sh to remove `nemsio_*.fd` symlinks; nemsio codes now gone in UFS_UTILS component version
9. version file updates to `hera.ver` and `orion.ver` to add new obsproc/prepobs run versions, update `hpc_ver` to 1.2.0, and add verson overrides for module versions that differ compared to WCOSS2

Some of the above build changes are to support the fact that hpc-stack was installed differently on the R&D machines compared to WCOSS2 (dynamic instead of static).

The above updates do not impact build or run functionality on WCOSS2. The above has been tested (build and run) in cycled mode on WCOSS2 to confirm no functionality impact.

Note: there are two component versions uncommitted but tested alongside the above: ufs-weather-model `modulefileRnD` branch provided by @junwang-noaa and UFS_UTILS `hotfix/gfsv16.2.0` branch provided by @GeorgeGayno-NOAA. Can't build out of the box without those two component updates.

**Type of change**

- New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

These changes have been tested on Hera, Orion, and WCOSS2 to confirm checkout and build happen correctly. Also tested in cycled mode on WCOSS2 to confirm no impact on functionality there. Cycled tests still to be done on Hera/Orion.

Refs: #665

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
@KateFriedman-NOAA If you think my hotfix branch is working, let me create a formal tag. I don't want you pointing to that branch.

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA If you think my hotfix branch is working, let me create a formal tag. I don't want you pointing to that branch.

@GeorgeGayno-NOAA Yes, I did some cycled testing with that hotfix branch on WCOSS2 and started cycled testing on Hera/Orion with that branch and it's working thus far. Please cut me a tag now and should I hit issues I'll let you know. Thanks!

**Comment by junwang-noaa** [comment]
@aliabdolali @JessicaMeixner-NOAA May I ask if the ww3 changes for ufs-weather-model are ready? If yes, I can update the ufs-weather-model branch modulefileRnD and merge the code changes to production/GFSv16. Thanks

**Comment by JessicaMeixner-NOAA** [comment]
@junwang-noaa I'm taking care of the operational issues for WW3 for now. There are no planned updates for WW3 for 16.2, if changes are required we were going to target them for 16.3, unless something has changed since our last emails/meetings. In terms of testing the updates for 16.3, I've heard no updates on hpc-stack requiring testing of potential changes or not.

**Comment by KateFriedman-NOAA** [comment]
@junwang-noaa @JessicaMeixner-NOAA @arunchawla-NOAA Correct me if I'm wrong but I believe we agreed to not support waves on Hera/Orion for this version of v16.2 but will continue to support waves on WCOSS2 in this version. This is all for the dev_v16 branch and will not be going into the feature/ops-wcoss2 branch or operations branch for v16.2.  The dev_v16 branch is the base for v16.3+ upgrades post-WCOSS2 go-live but needs to support v16.2 to start before we apply v16.3 updates.

Then the WW3 updates would be made for the v16.3 system and will be supported on all three platforms via the soon-to-be-made g-w branch for the v16.3 upgrade (November).

**Comment by JessicaMeixner-NOAA** [comment]
@KateFriedman-NOAA I think you are correct.

**Comment by junwang-noaa** [comment]
@KateFriedman-NOAA Thanks for clarifying. I thought nceplibs team will update the library so that the model branch will support ww3 on hera/orion, thanks for confirming that is not the case. Also can you clarify what are the dev_v16 branch,  feature/ops-wcoss2 branch, the operations branch for v16.2 and the g-w branch? Is it correct that the model branch with GFSv16 atm only capability on hera/orion will not be updated with future ww3 updates?

**Comment by KateFriedman-NOAA** [comment]
> I thought nceplibs team will update the library so that the model branch will support ww3 on hera/orion, thanks for confirming that is not the case.

Good point @junwang-noaa let me check on this... @arunchawla-NOAA was a request to reinstall the hpc-stack-gfsv16 statically on Hera/Orion submitted?

**Comment by arunchawla-NOAA** [comment]
I dropped the ball on that. Let me see about that

**Comment by KateFriedman-NOAA** [comment]
> Is it correct that the model branch with GFSv16 atm only capability on hera/orion will not be updated with future ww3 updates?

@junwang-noaa Correct. No more WW3 code changes should be coming into the GFSv16.2 ufs-weather-model production branch. Support for Hera/Orion should stick to module/build updates and no code updates. Code updates to ufs-weather-model and/or WW3 to support Hera/Orion should be held for the future v16.3 upgrade (which will start testing in a month or so).

> Also can you clarify what are the dev_v16 branch, feature/ops-wcoss2 branch, the operations branch for v16.2 and the g-w branch?

The `feature/ops-wcoss2` branch is the current GFSv16.2 package for WCOSS2 and will be merged into the `operations` branch around go-live (a.k.a. implementation date). That feature branch will then go away.

The `dev_v16` branch will ingest any final updates from `feature/ops-wcoss2` and `operations` branches for v16.2, be what supports v16.2 on all three platforms, and be the base for future GFSv16+ upgrades. The GFSv16.3 global-workflow branch will be cut from this branch (name not set yet but maybe `release/gfs.v16.3.0`). Then that v16.3 global-workflow branch will be merged into the `operations` branch when that version implements on WCOSS2. The `dev_v16` branch will stick around to support the v16+ system on WCOSS2/Hera/Orion as needed but may also go away after v16.3 goes into ops. It's future is TBD but it will most likely go away as we prep the GFSv17+ hand-off branch. Since it's not easy to get bug fixes into ops to also support R&D we have to maintain this separate branch.

In case I confused you..."g-w" is just my short-hand for "global-workflow". :)

Hope that helps! It's confusing but since we have so many overlapping implementations/supports this is how we're keeping things sorted right now.

File: sorc/enkf_chgres_recenter_nc.fd/makefile
Comment (WalterKolczynski-NOAA) [inline sorc/enkf_chgres_recenter_nc.fd/makefile]: Remove dead code?

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Approved pending testing, though there is one piece of commented code that could be removed.

**Comment by junwang-noaa** [comment]
@KateFriedman-NOAA Thanks for the clarification. Since there is not source code change in the modulefileRnD branch, once it is confirmed no ww3 changes will not be added, I will merge the branch to model production/GFS.v16 branch and create a tag. Thanks

**Comment by JessicaMeixner-NOAA** [comment]
@junwang-noaa there will be no WW3 changes for this PR. We will continue to focus any potential WW3 changes for 16.3

**Comment by KateFriedman-NOAA** [comment]
Thanks for confirming @JessicaMeixner-NOAA ! 

@junwang-noaa sounds like we're ready for you to "merge the branch to model production/GFS.v16 branch and create a tag". I've tested the build of your `modulefileRnD` branch on WCOSS2  with WW3=Y and Hera/Orion with WW3=N (no issues observed) and I have run fcst jobs with it on all three machines without issues. Thanks!

**Comment by KateFriedman-NOAA** [comment]
@WalterKolczynski-NOAA @aerorahul I will hold this PR until I get the UFS_UTILS and ufs-weather-model tags discussed. Will address other comments while waiting for that. 

There will be further PRs as I perform cycled testing on Hera/Orion but this branch is working well on WCOSS2 thus far (mostly out of the box, minus the two components awaiting tagging). I anticipate some time to sort out obsproc on Hera/Orion, in addition to resource optimization on Hera/Orion, and potential other updates as I make my way through all cycle jobs. Updates for that will be in future PRs for this branch.

File: sorc/enkf_chgres_recenter_nc.fd/makefile
Comment (aerorahul) [inline sorc/enkf_chgres_recenter_nc.fd/makefile]: please delete commented line

**Review by aerorahul — COMMENTED** [review]
Changes look good to me.
The proof will however be in the pudding when this is built and run.

**Comment by junwang-noaa** [comment]
@KateFriedman-NOAA The model tag: GFS.v16.2.1 was created. Thanks

File: sorc/enkf_chgres_recenter_nc.fd/makefile
Comment (KateFriedman-NOAA) [reply sorc/enkf_chgres_recenter_nc.fd/makefile]: removed, thanks for catching!

Comment (KateFriedman-NOAA) [reply sorc/enkf_chgres_recenter_nc.fd/makefile]: removed, thanks for catching!

**Comment by KateFriedman-NOAA** [comment]
> The model tag: GFS.v16.2.1 was created. Thanks

Thanks @junwang-noaa ! 

> https://github.com/ufs-community/UFS_UTILS/issues/629#issuecomment-1082378409

New UFS_UTILS tag cut: [ops-gfsv16.2.1](https://github.com/ufs-community/UFS_UTILS/releases/tag/ops-gfsv16.2.1). Thanks @GeorgeGayno-NOAA !

Tested build of both new tags on WCOSS2, Hera, and Orion...no issues. Ran ufs-weather-model build with `WW3=Y` on WCOSS2 and `WW3=N` on Hera/Orion. Updated build_fv3.sh to set this based on machine for this version of the GFS. Committed to branch @ [fdd4fcd](https://github.com/NOAA-EMC/global-workflow/pull/700/commits/fdd4fcdcd33143bb299aca45c94f80914e3a667b).

I have updated checkout.sh with these two tags @ [c52e4b8](https://github.com/NOAA-EMC/global-workflow/pull/700/commits/c52e4b8b249430b2bac5f461d155e79f72f7fd5d). Also included needed "--init" flag with GSI submodule checkout (resolves clone issue after libsrc submodule update, forgot to commit from WCOSS2 clone earlier).

**Comment by KateFriedman-NOAA** [comment]
Any further questions/comments on this PR before it is submitted? Reminder, this is just the build and version files updates...which have been tested successfully during the build steps on all three machines. Additional updates will be sent back as cycled testing occurs with this branch on Hera, Orion, and WCOSS2.

@aerorahul Anything blocking your approval on this PR that I can address? Thanks!

**Review by aerorahul — APPROVED** [review]
looks good.

**Pull request merged**
